src/v/crypto/BUILD

```diff
@@ -0,0 +1,106 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+load("//bazel:test.bzl", "redpanda_cc_gtest")
+
+redpanda_cc_library(
+    name = "crypto",
+    srcs = [
+        "crypto.cc",
+        "digest.cc",
+        "hmac.cc",
+        "internal.h",
+        "key.cc",
+        "key.h",
+        "ossl_context_service.cc",
+        "random.cc",
+        "signature.cc",
+        "ssl_utils.cc",
+        "ssl_utils.h",
+    ],
+    hdrs = [
+        "include/crypto/crypto.h",
+        "include/crypto/exceptions.h",
+        "include/crypto/ossl_context_service.h",
+        "include/crypto/types.h",
+    ],
+    exclude_layering_check = True,
```

Comment:
Is open SSL to blame here?

Reply:
No, I think it is foreign_cc_rules itself. Layering checks are disabled on foreign_cc_rules, but from what I can tell, the library targets produced by the foreign_cc_rules don't seem to encode the header files in a way that allow the clang modules to be created.

i'm tempted to try to create some kind of alias library that would be a hacky fix.

engaged with bazel folks here https://github.com/bazelbuild/rules_foreign_cc/issues/1221

Reply:
https://redpandadata.atlassian.net/browse/CORE-5572